### PR TITLE
[TS] LPS-191780 Use portal class loader to load in test-portal-ext.properties

### PR DESCRIPTION
**Pull request by DaneszLR**
Hi Team,

https://liferay.atlassian.net/browse/LPS-191780

Issue:
Customer wanted to overwrite company.web.id and do an integration test. The test read up the portal-ext.properties file and set the company.web.id to the new value, but right before the test, the test-portal-impl.properties file got read up overwriting the company.web.id. Using test-portal-impl-ext.properties could resolve the issue, but it doesn't get read up, only from portal-test.jar. In 6.2 it was possible to put the -ext file outside the jar, but since then test-portal-impl.properties was moved from portal-impl in LPS-59148 to portal-test module so that we don't have to force every test module to Include-Resource it.

Customer stated putting it in the jar doesn't solve their problem because the jar gets rebuilt and the added properties file gets lost.

A slack thread about it: https://liferay.slack.com/archives/GLCFMMZQW/p1691486956999219

Solution:
For reading up the -ext file use portal class loader, so it can be reached from webapps/ROOT/WEB-INF/classes

Testing:
Tested it locally. The solution works.

Please review my PR!

Best regards,
Dani

### CODE REVIEW

**Comment by catarineaguiar** [comment]
ci:test:relevant

**Comment by catarineaguiar** [comment]
ci:test:sf

**Comment by Kvale1733** [comment]
ci:test:stable

**Comment by Kvale1733** [comment]
ci:test:stable

**Comment by Kvale1733** [comment]
@DaneszLR Regression found. There are several integration tests failing for the same error.

Such as com.liferay.portal.dao.db.test.DBTest
```
 [exec] com.liferay.portal.dao.db.test.DBTest > initializationError STARTED
     [exec] 
     [exec] com.liferay.portal.dao.db.test.DBTest > initializationError FAILED
     [exec]     java.lang.ExceptionInInitializerError
     [exec]         at com.liferay.portal.kernel.test.util.TestPropsValues.<clinit>(TestPropsValues.java:33)
     [exec]         at com.liferay.portal.kernel.test.rule.TimeoutTestRule.<clinit>(TimeoutTestRule.java:25)
     [exec]         at com.liferay.portal.test.rule.LiferayIntegrationTestRule._getTestRules(LiferayIntegrationTestRule.java:34)
     [exec]         at com.liferay.portal.test.rule.LiferayIntegrationTestRule.<init>(LiferayIntegrationTestRule.java:27)
     [exec]         at com.liferay.portal.dao.db.test.DBTest.<clinit>(DBTest.java:49)
```

```
[exec]         java.lang.NullPointerException
     [exec]             at com.liferay.portal.kernel.test.util.TestPropsUtil.<init>(TestPropsUtil.java:54)
     [exec]             at com.liferay.portal.kernel.test.util.TestPropsUtil.<clinit>(TestPropsUtil.java:107)
```
[Log](https://testray.liferay.com/reports/production/logs/2023-08/test-1-40/test-portal-acceptance-pullrequest(master)/5258/modules-integration-mysql57-jdk8_stable/0/1/jenkins-console.txt.gz?authuser=0)

I wasn't able to reproduce locally. Tried using these steps also: https://liferay.atlassian.net/wiki/spaces/~85516578/pages/2369454549/Integration+Tests but couldn't reproduce that way either. 

But these tests are failing consistantly on CI on this PR.

**Comment by DaneszLR** [comment]
Hi @Kvale1733,
Thank you for the CI and local test. I will look into the issue.

File: portal-test/src/com/liferay/portal/kernel/test/util/TestPropsUtil.java [hunk 1]
```diff
@@ -49,8 +50,8 @@ private TestPropsUtil() {
 			ReflectionUtil.throwException(ioException);
 		}
 
-		try (InputStream inputStream = TestPropsUtil.class.getResourceAsStream(
-				"/test-portal-impl-ext.properties")) {
+		try (InputStream inputStream = PortalClassLoaderUtil.getClassLoader()
+			.getResourceAsStream("/test-portal-impl-ext.properties")) {
```
Comment (tinatian) [inline portal-test/src/com/liferay/portal/kernel/test/util/TestPropsUtil.java:54]: @DaneszLR Please do not chain the method. You should do 
ClassLoader classloader = PortalClassLoaderUtil.getClassLoader();
classloader..getResourceAsStream(

**Comment by DaneszLR** [comment]
Hi @tinatian ,
The PR has been resent at https://github.com/liferay-core-infra/liferay-portal/pull/5366

**Pull request closed without merging**